### PR TITLE
Add core Button Component

### PR DESCRIPTION
**Pull request by sliptype**
### Description

* Moves the `Button` from profile screen into common
* Adds a `size` prop
* Changes `iconLeft` to `icon` and adds an `iconPosition` prop
* Marks the existing `Button` as deprecated
* Uses the new `Button` in a few places

Some of the stems Button variants don't match what's currently in the app, going to defer to stems and opt for reuse. Also the stems Button sizes need some work, will dig into that further

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.

### How will this change be monitored?

For features that are critical or could fail silently please describe the monitoring/alerting being added.

### CODE REVIEW

File: packages/mobile/src/components/core/Button.tsx [hunk 1]
```diff
@@ -65,64 +66,126 @@ const useStyles = makeStyles(
       common: variantStyles.primary
     }
 
+    const sizeStyles = {
```
Comment (dylanjeffers) [inline packages/mobile/src/components/core/Button.tsx:69]: this is great!

File: packages/mobile/src/components/core/Button.tsx [hunk 2]
```diff
@@ -65,64 +66,126 @@ const useStyles = makeStyles(
...
+        ...flexRowCentered()
       },
       text: {
-        fontSize: 11,
         fontFamily: typography.fontByWeight.bold,
-        textTransform: 'uppercase'
-      },
-      leftIcon: {
-        marginRight: spacing(1)
+        letterSpacing: 0.5
```
Comment (dylanjeffers) [inline packages/mobile/src/components/core/Button.tsx:142]: nice

File: packages/mobile/src/components/core/Button.tsx [hunk 3]
```diff
@@ -65,64 +66,126 @@ const useStyles = makeStyles(
...
+    variant?: 'primary' | 'secondary' | 'common'
   }
 
 export const Button = (props: ButtonProps) => {
   const {
-    title,
-    iconLeft: IconLeft,
-    variant,
+    icon: Icon,
+    iconPosition = 'right',
```
Comment (dylanjeffers) [inline packages/mobile/src/components/core/Button.tsx:174]: cool didn't know this was the default pos

File: packages/mobile/src/components/enable-push-notifications-drawer/EnablePushNotificationsDrawer.tsx [hunk 4]
```diff
@@ -175,11 +174,10 @@ const EnablePushNotificationsDrawer = () => {
         <Button
           title={messages.enable}
           onPress={enablePushNotifications}
-          containerStyle={styles.button}
+          style={styles.button}
```
Comment (dylanjeffers) [inline packages/mobile/src/components/enable-push-notifications-drawer/EnablePushNotificationsDrawer.tsx:177]: up to you on if we do this this pr, but ive seen a lot of design systems have a `fullWidth` prop, that adds `width: 100%`

File: packages/mobile/src/screens/track-screen/TrackScreenDownloadButtons.tsx [hunk 5]
```diff
@@ -83,24 +68,16 @@ const DownloadButton = ({
...
-        isProcessing ? (
-          <LoadingSpinner
-            color={fill as string}
-            style={{ height: 18, width: 18 }}
-          />
-        ) : (
-          <IconDownload fill={fill} height={18} width={18} />
-        )
-      }
+      variant='common'
```
Comment (dylanjeffers) [inline packages/mobile/src/screens/track-screen/TrackScreenDownloadButtons.tsx:71]: thanks for adding common variant

File: packages/mobile/src/screens/track-screen/TrackScreenRemixes.tsx [hunk 6]
```diff
@@ -93,13 +80,13 @@ export const TrackScreenRemixes = ({
...
-        type={ButtonType.SECONDARY}
+        icon={IconArrow}
+        variant='primary'
+        size='medium'
         onPress={goToAllRemixes}
-        style={styles.button}
-        containerStyle={styles.buttonContainer}
-        textStyle={styles.buttonText}
-        iconStyle={styles.buttonIcon}
+        styles={{
```
Comment (dylanjeffers) [inline packages/mobile/src/screens/track-screen/TrackScreenRemixes.tsx:87]: could just be style, but nbd

**Review by dylanjeffers — APPROVED** [review]
this looks really clean, thanks a bunch for moving this over

File: packages/mobile/src/components/enable-push-notifications-drawer/EnablePushNotificationsDrawer.tsx (hunk 4 above)
Comment (sliptype) [reply packages/mobile/src/components/enable-push-notifications-drawer/EnablePushNotificationsDrawer.tsx:177]: Let's discuss this more! I can see an argument against having `fullWidth` as a prop too

Comment (sliptype) [reply packages/mobile/src/components/enable-push-notifications-drawer/EnablePushNotificationsDrawer.tsx]: Nvm, I went ahead and added it bc I realized there are 2 places where width 100 needs to be set

**Pull request merged**